app/src/main/java/io/intrepid/contest/screens/entrysubmission/entryname/EntryNamePresenter.java

```diff
@@ -0,0 +1,41 @@
+package io.intrepid.contest.screens.entrysubmission.entryname;
+
+import android.support.annotation.NonNull;
+
+import io.intrepid.contest.base.BasePresenter;
+import io.intrepid.contest.base.PresenterConfiguration;
+
+import static io.intrepid.contest.screens.entrysubmission.entryname.EntryNameContract.Presenter;
+import static io.intrepid.contest.screens.entrysubmission.entryname.EntryNameContract.View;
+
+public class EntryNamePresenter extends BasePresenter<View> implements Presenter {
+
+    public EntryNamePresenter(@NonNull View view,
+                              @NonNull PresenterConfiguration configuration) {
+        super(view, configuration);
+    }
+
+    @Override
+    public void onViewCreated() {
+        super.onViewCreated();
+        view.onContestNameRetrieved("Chili Cookoff");
+    }
+
+    @Override
+    public void onEntryNameTextViewFocusChanged(boolean isFocused) {
+        if (isFocused) {
+            view.showEntryNameNextButton();
+        } else {
+            view.hideEntryNameNextButton();
+        }
+    }
+
+    @Override
+    public void onEntryNameTextViewTextChanged(CharSequence newText) {
+        if (newText.length() == 0) {
+            view.disableEntryNameNextButton();
+        } else if (newText.length() == 1) {
```

Comment:
Does android check if the visibility is already "VISIBLE" before applying the changes? Or is the impact negligible to set it again?